ssl/statem/statem_clnt.c

```diff
@@ -1828,9 +1828,10 @@ MSG_PROCESS_RETURN tls_process_server_certificate(SSL *s, PACKET *pkt)
     if (!SSL_IS_TLS13(s)) {
         exp_idx = ssl_cipher_get_cert_index(s->s3->tmp.new_cipher);
         if (exp_idx >= 0 && i != exp_idx
-            && (exp_idx != SSL_PKEY_GOST_EC ||
-                (i != SSL_PKEY_GOST12_512 && i != SSL_PKEY_GOST12_256
-                 && i != SSL_PKEY_GOST01))) {
+                && (exp_idx != SSL_PKEY_ECC && i != SSL_PKEY_ED25519)
```

Comment:
This doesn't look right to me. Should this be:

    (exp_idx != SSL_PKEY_ECC ||  i != SSL_PKEY_ED25519)